src/Squisher.js

```diff
@@ -17,7 +19,39 @@ class Squisher {
         this.listeners.remove(listener);
     }
 
-    initialize() {
+    async initialize() {
+        const gameAssets = this.game.getAssets ? this.game.getAssets() : [];
+        
+        let assetBundleSize = 0;
+
+        for (const key in gameAssets) {
+            const payload = await gameAssets[key].getData();
+            const assetKeyLength = 32;
+            let keyIndex = 0;
+            const assetKeyArray = new Array(32);
+            while (keyIndex < assetKeyLength && keyIndex < key.length) {
+                assetKeyArray[keyIndex] = key.charCodeAt(keyIndex);
+                keyIndex++;
+            }
+
+            const encodedLength = (payload.length + assetKeyLength).toString(36);
+            
+            const assetType = gameAssets[key].info.type === "image" ? 1 : 2;
+
+            this.assets[key] = [ASSET_TYPE, assetType, encodedLength.charCodeAt(0), encodedLength.charCodeAt(1), encodedLength.charCodeAt(2), encodedLength.charCodeAt(3), ...assetKeyArray, ...payload];
```

Comment:
This logic should be abstracted. Pretty sure we won't support video in the near future (though I suppose that's not too crazy), so image/audio are probably enough for assets for now